addons/mail/models/mail_channel.py

```diff
@@ -491,10 +500,24 @@ def _channel_message_notifications(self, message, message_format=False):
         message_format = message_format or message.message_format()[0]
         notifications = []
         for channel in self:
-            notifications.append([(self._cr.dbname, 'mail.channel', channel.id), dict(message_format)])
+            notifications.append((
+                channel,
+                'mail.channel_new_message',
+                {
+                    'id': channel.id,
```

Comment:
payload: for these 2 I think the channel id might be useless, because model/res_id is already part of the message format. The payload could directly be the message_format itself, with no subkey